source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -165,3 +165,330 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
     m_hAlignment = value;
 }
 
+bool TextBlock::scanForISO8601(std::string::const_iterator &itr, std::string::const_iterator &end, 
```

Comment:
>scanForISO8601 [](start = 16, length = 14)

Any specific reason why you take the iterators instead of the std::string?

Reply:
i wanted to keep moving forward with text process.

---
In reply to: [148396093](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148396093) [](ancestors = 148396093)

Reply:
Sorry, this comment was meant for the Parse method. You always call it with the string's begin() and end() iterators.

---
In reply to: [148613744](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148613744) [](ancestors = 148613744,148396093)

Reply:
sorry, i'm not following you; can you explain?

---
In reply to: [148614839](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148614839) [](ancestors = 148614839,148613744,148396093)

Reply:
The parseISO8601 method is always called with the string's begin() and end() iterators. That function should probably just take the string instead of begin and end iterators

---
In reply to: [148615412](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148615412) [](ancestors = 148615412,148614839,148613744,148396093)

Reply:
that's very good point.
yup, that's correct.
thanks


---
In reply to: [148616076](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148616076) [](ancestors = 148616076,148615412,148614839,148613744,148396093)